src/chi/javascript/tooltip.js

```diff
@@ -0,0 +1,170 @@
+    };
+    this._focusEventHandler = function() {
+      self._focused = true;
+      if (!self._shown) {
+        self.show();
+      }
+    };
+    this._blurEventHandler = function() {
+      self._focused = false;
+      if (self._shown && !self._hovered) {
+        self.hide();
+      }
+    };
+
+    this._elem.addEventListener('mouseover',this._mouseOverEventHandler,false);
+    this._elem.addEventListener('mouseout', this._mouseOutEventHandler, false);
+    this._elem.addEventListener('focus',this._focusEventHandler,false);
+    this._elem.addEventListener('blur', this._blurEventHandler, false);
+
+  }
+
+  show() {
+    this._shown = true;
+    Util.addClass(this._tooltipElem, CLASS_ACTIVE);
+    const transition = this._tooltipElem.style.transition;
+    this._tooltipElem.style.transition = 'none';
+    this._tooltipElem.style.transform = this._preAnimationTransformStyle;
+    this._tooltipElem.style.opacity = '0';
+    let self = this;
+    window.setTimeout(function(){
+      self._tooltipElem.style.transition = transition;
+      self._tooltipElem.style.transform = self._postAnimationTransformStyle;
+      self._tooltipElem.style.opacity = '1';
+      self._tooltipElem.setAttribute('aria-hidden', 'false');
+    },0);
+  }
+
+  hide() {
+    this._shown = false;
+    Util.removeClass(this._tooltipElem, CLASS_ACTIVE);
+    let self = this;
+    window.setTimeout(function(){
+      self._tooltipElem.style.transform = self._preAnimationTransformStyle;
+      self._tooltipElem.style.opacity = '0';
+      self._tooltipElem.setAttribute('aria-hidden', 'true');
+    },0);
+  }
+
+  _createTooltip () {
+    this._tooltipElem = document.createElement('div');
+    this._tooltipElem.setAttribute('class', 'a-tooltip');
+    this._tooltipElem.id = COMPONENT_TYPE + Util.getData(this._elem, COMPONENT_TYPE);
+    this._tooltipElem.innerText = this._elem.dataset.tooltip;
+    this._elem.parentNode.appendChild(this._tooltipElem);
+    this._elem.setAttribute('aria-describedby', this._tooltipElem.id);
+    this._tooltipElem.setAttribute('aria-hidden', 'true');
+
+
+    let self = this;
+
+    this._savePopperData = function (data) {
+      self._popperData = data;
+      self._preAnimationTransformStyle = null;
+      self._postAnimationTransformStyle = data.styles.transform;
+      if (data.placement.indexOf("top") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left}px, ${data.popper.top + 20}px, 0px)`;
+      } else if (data.placement.indexOf("right") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left + 20}px, ${data.popper.top}px, 0px)`;
+      } else if (data.placement.indexOf("bottom") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left}px, ${data.popper.top - 20}px, 0px)`;
+      } else if (data.placement.indexOf("left") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left - 20}px, ${data.popper.top+20}px, 0px)`;
```

Comment:
Left and right have different top values. Can we update the left to render the same as the right?

Reply:
To clarify, let's remove the +20 from left